paper_framework_sources/p2079_system_execution_context.md

```diff
@@ -337,6 +390,61 @@ This would work at task granularity, for each `schedule()` call that we connect
+* As the default implementation is created outside of the host part, it can be shared between multiple binaries in the same process.
+* IN PROGRESS: Provides an example on implementing the `drive()` behavior by replacing the default implementation.
+* TODO: Use OS scheduler for implementation.
+
+## Addressing received feedback
+
+### Allow for system context to borrow threads
+Early feedback on the paper from Sean Parent suggested a need for the system context to carry no threads of its own and to take over the main thread.
+While in [@P2079R2] we proposed `execute_chunk` and `execute_all`, these enforce a particular implementation on the underlying execution context.
+Instead, we simplify the proposal by removing this functionality and assuming that it is implemented by link-time or compile-time replacement of the context.
+We assume that the underlying mechanism to drive the context, should one be necessary, is implementation-defined.
+This allows for custom hooks into an OS thread pool, or a simple `drive()` method in main.
+
+### Allow implementations to use Grand Central Dispatch and Windows Thread Pool
+
+In the current form of the paper, we allow implementations to define the best choice for implementing the system context for a particular system.
+This includes using Grand Central Dispatch on Apple platforms and Windows Thread Pool on Windows.
+
+In addition, we encourage implementations to allow the replaceability of the system context implementation.
+This means that users should be allowed to write their own system context implementations that depend on OS facilities.
+
+### Priorities and elastic pools
+
+Feedback from Sean Parent:
+> There is so much in that proposal that is not specified. What requirements are placed on the system scheduler? Most system schedulers support priorities and are elastic (i.e., blocking in the system thread pool will spin up additional threads to some limit).
+
+The lack of details in the specification is intentional, allowing implementers to make the best compromises for each platform.
+As different platforms have different needs, constraints, and optimization goals, the authors believe that it is in the best interest of the users to leave some of these details as Quality of Implementation (QOI) details.
+
+
+
+# Questions to ask LEWG
+
+## Do we want to standardize the "replaceability" aspect of `system_context` (as opposed to leaving this to be implementation defined)?
+
+Proposed answer: NO.
+We want to allow vendors the flexibility of choosing the best implementation.
+At the same time, the authors of this paper are willing to contribute to creating an informal basis for implementing replaceability in a common way among major vendors.
+
+## Do we want to add explicit support for `drive()` functionality in `system_context` (as opposed to letting this be implementation-defined)?
+
+Proposed answer: NO.
+We would prefer this to be implementation-defined.
```

Comment:
I would word this more strongly. It isn't so much that it should be implementation-defined, it's that for the standard context in all likelihood it makes no sense. It only makes sense as functionality for some thread-limited situation, where the context will run thread on a single application event loop, and those hooks are all very implementation-defined.

Reply:
Some uses of system_context are going to have to use the main_context if the system really only has a single thread. So replacing system with main is still going to be necessary. At that point I'm not sure I see the value in main being queryable, rather than an implementation. Obviously, I missed the nuance of the discussion.

Should we simply update the wording here to say it is going to be a separate paper, and doesn't affect the core definition here.